src/group_cache.rb

```diff
@@ -29,6 +29,12 @@ module Metalware
   class GroupCache
     include Enumerable
 
+    def self.update
```

Comment:
`update` as a name for this could maybe be clearer, as it's not clear that this should take a block and won't update the cache immediately - how about `transaction`, as it functions similarly to a database transaction? I.e. it will either save the cache if everything succeeds or not save it (rollback to the previous state) if any error occurs.